ts/sagas/onboarding.ts

```diff
@@ -97,11 +105,42 @@ function* tosCheckSaga(): Iterator<Effect> {
     });
     yield put(navigateToOnboardingTosScreenAction);
 
-    // Here we wait the user accept the ToS
-    yield take(TOS_ACCEPT_REQUEST);
+    // Loop until this step is fulfilled: TOS accepted and if necessary Profile upserted.
+    while (true) {
+      // Here we wait the user accept the ToS
+      yield take(TOS_ACCEPT_REQUEST);
+
+      // Get the current user profile from the store
+      const userProfile: ProfileState = yield select(profileSelector);
+
+      // We have the profile info but the user hasn't a profile active on API
+      // NOTE: `has_profile` is a boolean that is true if the profile is active in the API
+      if (userProfile && !userProfile.has_profile) {
+        // Upsert the user profile to enable inbox and webhook
+        yield put(
+          profileUpsertRequest({
```

Comment:
shouldn't this also set the flag in the profile that stores the fact that the tos has been accepted?

Reply:
I don't think the API already implements it.

```
ExtendedProfile:
  description: |-
    Describes the citizen's profile, mostly interesting for preferences
    attributes.
  type: object
  properties:
    email:
      $ref: "#/EmailAddress"
    blocked_inbox_or_channels:
      $ref: "#/BlockedInboxOrChannels"
    preferred_languages:
      $ref: "#/PreferredLanguages"
    is_inbox_enabled:
      $ref: "#/IsInboxEnabled"
    is_webhook_enabled:
      $ref: "#/IsWebhookEnabled"
    version:
      type: integer
```